data-structures/filecoin/chain.md

```diff
@@ -0,0 +1,1225 @@
+  EarlyTerminated &BitFieldQueueAMT # AMT[ChainEpoch]BitField
+  LivePower MinerV0PowerPair
+  FaultyPower MinerV0PowerPair
+  RecoveringPower MinerV0PowerPair
+} representation tuple
+
+type MinerV0PowerPair struct {
+  Raw StoragePower
+  QA StoragePower
+} representation tuple
+```
+
+**v2**
+
+**AMT**: This is an ADL representing `type MinerV2PartitionList [MinerV2Partition]` indexed by `PartitionNumber`.
+
+```ipldsch
+type MinerV2PartitionAMT struct {
+  height Int
+  count Int
+  node MinerV2PartitionAMTNode
+} representation tuple
+
+type MinerV2PartitionAMTNode struct {
+  bitmap Bytes
+  children [&MinerV2PartitionAMTNode]
+  values [MinerV2Partition]
+} representation tuple
+
+type MinerV2Partition struct {
+  Sectors BitField
+  Unproven BitField
+  Faults BitField
+  Recoveries BitField
+  Terminated BitField
+  ExpirationsEpochs &MinerV0ExpirationSetAMT # AMT[ChainEpoch]ExpirationSet
+  EarlyTerminated &BitFieldQueueAMT # AMT[ChainEpoch]BitField
+  LivePower MinerV0PowerPair
+  UnprovenPower MinerV0PowerPair
+  FaultyPower MinerV0PowerPair
+  RecoveringPower MinerV0PowerPair
+} representation tuple
+```
+
+**AMT**: This is an ADL representing `type MinerV0ExpirationSetList [MinerV0ExpirationSet]` indexed by `ChainEpoch`.
+
+```ipldsch
+type MinerV0ExpirationSetAMT struct {
+  height Int
+  count Int
+  node MinerV0ExpirationSetAMTNode
+} representation tuple
+
+type MinerV0ExpirationSetAMTNode struct {
+  bitmap Bytes
+  children [&MinerV0ExpirationSetAMTNode]
+  values [MinerV0ExpirationSet]
+} representation tuple
+
+type MinerV0ExpirationSet struct {
+  OnTimeSectors BitField
+  EarlySectors BitField
+  OnTimePledge TokenAmount
+  ActivePower MinerV0PowerPair
+  FaultyPower MinerV0PowerPair
+} representation tuple
+```
+
+### MultisigActor
+
+```ipldsch
+type MultisigV0State struct {
+  Signers [Address]
+  NumApprovalsThreshold Int
+  NextTxnID TransactionID
+  InitialBalance TokenAmount
+  StartEpoch ChainEpoch
+  UnlockDuration ChainEpoch
+  PendingTxns &MultisigV0TransactionHAMT # HAMT[TransactionID]Multisigv0Transaction
+} representation tuple
+```
+
+**HAMT**: This is an ADL representing `type MultisigV0TransactionMap {TransactionIDString:MultisigV0Transaction}`. Where TransactionIDString is the varint (not uvarint) bytes (string) form of `TransactionID`.
+
+```ipldsch
+type MultisigV0TransactionHAMT struct {
+  map Bytes
+  data [ MultisigV0TransactionHAMTElement ]
+} representation tuple
+
+type MultisigV0TransactionHAMTElement union {
+  | MultisigV0TransactionHAMTLink "0"
+  | MultisigV0TransactionHAMTBucket "1"
+} representation keyed
+
+type MultisigV0TransactionHAMTLink &MultisigV0TransactionHAMT
+
+type MultisigV0TransactionHAMTBucket [ MultisigV0TransactionHAMTBucketEntry ]
+
+type MultisigV0TransactionHAMTBucketEntry struct {
+  key Bytes
+  value MultisigV0Transaction # inline
+} representation tuple
+
+type MultisigV0Transaction struct {
+  To Address
+  Value TokenAmount
+  Method MethodNum
+  Params Bytes
```

Comment:
these are the same params (cbor-encoding of one of the message params) as a message would have. Not sure if it's worth defining a "CborEncodedParams" byte type alias.